src/array/dictionary/mutable.rs

```diff
@@ -181,3 +193,63 @@ where
         Ok(())
     }
 }
+
+impl<K, M, T> TryPush<Option<T>> for MutableDictionaryArray<K, M>
+where
+    K: DictionaryKey,
+    M: MutableArray + TryPush<Option<T>>,
+    T: Hash,
+{
+    fn try_push(&mut self, item: Option<T>) -> Result<()> {
+        if let Some(value) = item {
+            if self.try_push_valid(&value)? {
+                self.values.try_push(Some(value))
+            } else {
+                Ok(())
+            }
+        } else {
+            self.push_null();
+            Ok(())
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
```

Comment:
Could we move this to `tests/it/array`? we have all our tests there so that the compilation of the crate is not triggered when only the tests change.

Reply:
I can do that. It will need two new `pub(crate)` functions on `MutableDictionaryArray`, which are required to test `TryPush`:

```rust

    /// Returns the dictionary map
    pub(crate) fn map(&self) -> &HashedMap<u64, K> {
        &self.map
    }

    pub(crate) fn keys(&self) -> &MutablePrimitiveArray<K> {
        &self.keys
    }
```

I'll also add an explicit return for the doctest to satisfy Miri.